cloudinit/stages.py

```diff
@@ -63,6 +75,132 @@
 )
 
 
+@dataclass
+class Semaphore:
+    semaphore: helpers.FileSemaphores
+    args: Tuple[str, str]
+
+
+class CacheMode(str, Enum):
+    trust = "trust"
+    check = "check"
+
+
+# used in status.json and some log strings
```

Comment:
This section defines a variety of magic strings that were previously scattered throughout main.py and stages.py.